lib/h_pyramid_sentry/setup.py

```diff
@@ -1,22 +1,88 @@
+import os
+
 from setuptools import setup, find_packages
 
-VERSION = "1.0.dev0"
+PACKAGE = "h_pyramid_sentry"
 
 INSTALL_REQUIRES = [
     'sentry-sdk',
     'pyramid',
     'pyramid_retry',
     'celery',
 ]
+
 TESTS_REQUIRE = INSTALL_REQUIRES + [
     'pytest',
     'coverage',
 ]
 
+
+def from_file(filename):
+    with open(filename) as fh:
+        return fh.read()
+
+
+def read_egg_version():
+    pkg_info_file = PACKAGE + '.egg-info/PKG-INFO'
+    if not os.path.isfile(pkg_info_file):
+        return None
+
+    with open(pkg_info_file) as fh:
+        for line in fh:
+            if line.startswith('Version'):
+                return line.strip().split('Version: ')[-1]
+
+
+def get_version(major_version, build_var='BUILD'):
+    # If we have a
+    build = os.environ.get(build_var)
+    if build:
+        return major_version + '.' + build
+
+    # We need to do this for source distributions, as setup.py is re-run when
+    # installed this way, and we would always get 'dev0' as the version
+    # Wheels and binary installs don't work this way and read from PKG-INFO
+    # for them selves
+    egg_version = read_egg_version()
+    if egg_version:
+        return egg_version
+
+    return major_version + '.dev0'
+
+
 setup(
-    name="h_pyramid_sentry",
-    description="A Pyramid plugin for Sentry to suppress errors",
-    version=VERSION,
+    # Metadata
+    # https://docs.python.org/3/distutils/setupscript.html#additional-meta-data
+
+    name=PACKAGE,
+    version=get_version(major_version='1.0'),
+    description="A Pyramid plugin for integrating Sentry logging",
```

Comment:
It doesn't really matter but I'd describe Sentry as error tracking rather than logging (we have a different app for logging)

Reply:
"Topic :: Software Development :: Bug Tracking" ?

Reply:
@lyzadanger chipped in with a vote for the monitoring option, which is good enough for me. I don't think it's important enough to tell a priest, so I've just gone ahead with that. Confusingly all of this is in the separate repo.